### ISSUE TITLE
Cannot instantiate a Mecab instance from a resource bundle path which contains a space

### ISSUE THREAD

**Issue by shirakaba** [shirakaba/mecab-ko#1]
This can happen if your `PRODUCT_NAME` (the "display name" for your app, e.g. `LinguaBrowse X` -> `LinguaBrowse X.app`) contains a space.

**Comment by dtc-ylvov** [comment]
Confirmed. Had to delete the space in PRODUCT_NAME. I hope to use it again someday :)

**Comment by shirakaba** [comment]
This bit me again.

The error can be seen in the debugger in the context of `tagger.cpp` in various places, but for some reason 

```
/*
 * The sequence goes something like:
 * [MecabObjC.m] mecab_new2([[@"--output-format-type=none --dicdir " stringByAppendingString:[NSString stringWithFormat:@"%@", dicDirPath]] UTF8String]);
 * [libmecap.cpp] MeCab::createTagger(arg);
 * [tagger.cpp] Tagger::createTagger -> ModelImpl::open -> 
 * [param.cpp] Param::open ->
 * [tagger.cpp] setGlobalError(tagger->what());
 */
```

In theory,  `mecab_strerror(mecab_t *mecab)` *should* expose `MeCab::getLastError();`, but for whatever reason, it's just returning a null byte rather than the error that is clearly has available internally.

The underlying cause is that `Param::open` (in `param.cpp`) is a very poor parser of CLI args. It doesn't respect escapes, and it doesn't respect quotes around the args. So this is actually a problem with MeCab itself rather than this library. However, my C++ is far too weak to feel comfortable tackling this (and I'd rather not make any changes to the underlying source regardless).

The way I worked around this was to make this patch to stop using the `mecab_new2` convenience function:

## Before

```objc
- (instancetype)initWithDicDirPath:(NSString *)dicDirPath
{
    if(self = [super init])
    {
        
        mecab = mecab_new2([[@"--output-format-type=none --dicdir " stringByAppendingString:[NSString stringWithFormat:@"%@", dicDirPath]] UTF8String]);
        if (mecab == NULL) {
            fprintf(stderr, "error in mecab_new2: %s\n", mecab_strerror(NULL));
            return nil;
        }
    }
    return self;
}
```

## After

```objc
- (instancetype)initWithDicDirPath:(NSString *)dicDirPath
{
    if(self = [super init])
    {
        
        // Note: you could also specify where mecabrc lives by passing in --rcfile
        NSArray<NSString *> *argv = @[@"--output-format-type=none", @"--dicdir", dicDirPath];
        
        const char* testing[argv.count];
        for (int i = 0; i < argv.count; ++i) {
            testing[i] = [argv[i] UTF8String];
        }
        
        mecab = mecab_new((int)argv.count, (char **)testing);
        if (mecab == NULL) {
            fprintf(stderr, "error in mecab_new2: %s\n", mecab_strerror(NULL));
            return nil;
        }
    }
    return self;
}
```

I'll try to release a patch for that now.

**Comment by shirakaba** [comment]
Confirmed to be fixed in v0.4.1! I've released to both CocoaPods and npm under that same version (as always).
